src/Support/Field.php

```diff
@@ -243,11 +243,18 @@ protected function originalResolver(): ?Closure
      * @param array<int,mixed> $arguments
      * @param array<string,mixed> $fieldsAndArguments
      */
-    private function instanciateSelectFields(array $arguments, array $fieldsAndArguments): SelectFields
+    protected function instanciateSelectFields(array $arguments, array $fieldsAndArguments): SelectFields
     {
         $ctx = $arguments[2] ?? null;
 
-        return new SelectFields($this->type(), $arguments[1], $ctx, $fieldsAndArguments);
+        $selectFieldsClass = $this->selectFieldClass();
+
+        return new $selectFieldsClass($this->type(), $arguments[1], $ctx, $fieldsAndArguments);
+    }
+
+    protected function selectFieldClass(): string
+    {
+        return SelectFields::class;
```

Comment:
boom

This means we're agreeing and opening the `SelectFields` class to the world and encourage extending from it.

I think given the state of complexity and (missing) test coverage of that class, It's not unreasonable to say this is probably my worst nightmare 😄 

---
I'm in favour of the PR, but let me sort out the other issues first please; will get back to you.

Reply:
😅

Reply:
Np, my plan in the long run is to not hack less in this file and instead make prs that might benefit others. But right now this would help me to more easily see the actually changes I have made from the code I just copied over because of dependencies.